spec/models/breakeven_table_question_spec.rb

```diff
@@ -0,0 +1,189 @@
+      :fixed_costs,
+      :total_fixed_costs,
+      :total_fixed_costs_rampup,
+      :net_margin,
+      :net_margin_rampup,
+      :periods,
+      :units,
+    ].each do |row|
+      it "calculates #{row}" do
+        expect(subject.report[row]).to eq results[row]
+      end
+    end
+
+    [1, 2, 3].each do |product_number|
+      context "with Product #{product_number}" do
+        subject do
+          question = BreakevenTableQuestion.new(json)
+          question.report[:revenue].find { |p| p[:name] == "Product #{product_number}" }
+        end
+
+        let(:expected_product) { results[:revenue].find { |p| p[:name] == "Product #{product_number}" } }
+
+        it "revenue rampup" do
+          expect(subject[:rampup]).to eq expected_product[:rampup]
+        end
+      end
+    end
+  end
+
+  context 'without periods' do
+    let(:json) do
+      {
+        'products': [
+          { 'name': 'Product 1', 'description': 'Description', 'unit': 'Widgets', 'price': 100, 'cost': 50, 'quantity': 800 },
+          { 'name': 'Product 2', 'description': 'Description', 'unit': 'Stamp books', 'price': 120, 'cost': 60, 'quantity': 300 },
+          { 'name': 'Product 3', 'description': 'Description', 'unit': 'Flin Flons', 'price': 150, 'cost': 70, 'quantity': 100 },
+        ],
+        'fixed_costs': [
+          { 'name': 'Rent', 'amount': 1_5000 },
+          { 'name': 'Worker owners', 'amount': 28_000.0 },
+          { 'name': 'Employees', 'amount': 10_000.0 },
+          { 'name': 'Sales', 'amount': 10_000 },
+          { 'name': 'Utilities', 'amount': 2_000.0 },
+          { 'name': 'Insurance', 'amount': 1_000.0 },
+        ],
+      }
+    end
+
+    let(:results) do
+      {
+        revenue: [
+          { name: 'Product 1', quantity: 800.0, amount: 100.0, total: 80_000.0 },
+          { name: 'Product 2', quantity: 300.0, amount: 120.0, total: 36_000.0 },
+          { name: 'Product 3', quantity: 100.0, amount: 150.0, total: 15_000.0 },
+        ],
+        total_revenue: 131_000.0,
+        cogs: [
+          { name: 'Product 1', quantity: 800.0, amount: 50.0, total: 40_000.0 },
+          { name: 'Product 2', quantity: 300.0, amount: 60.0, total: 18_000.0 },
+          { name: 'Product 3', quantity: 100.0, amount: 70.0, total: 7_000.0 },
+        ],
+        total_cogs: 65_000.0,
+        gross_margin: 66_000.0,
+        fixed_costs: [
+          { name: 'Rent', amount: 15_000.0 },
+          { name: 'Worker owners', amount: 28_000.0 },
+          { name: 'Employees', amount: 10_000.0 },
+          { name: 'Sales', amount: 10_000 },
+          { name: 'Utilities', amount: 2_000.0 },
+          { name: 'Insurance', amount: 1_000.0 },
+        ],
+        total_fixed_costs: 66_000.0,
+        net_margin: 0.0,
+        periods: 1,
+        units: 'None'
+      }
+    end
+
+    subject { BreakevenTableQuestion.new(json) }
+
+    it 'calculates entire report properly' do
+      expect(subject.report).to eq results
+    end
+  end
```

Comment:
While those questions would have an impact on the rampup calculation I left those out as this is a new spec, wasn't already covered, and I was already bumping up against the end of the estimate.

@hooverlunch we have a couple solutions:

1. Take the time, and add new specs for above cases
2. Add pending specs for above cases
3. Ignore the extra edge cases, and fix if bugs arise.